### PR TITLE
Adds additional part to solution for extension: challenge 7

### LINKED ISSUE #4: Missing part of assignment.
https://github.com/FrontendMasters/fm-snippets/blob/0286522fd6f5aa0fa192759976b126f0ed3718d9/2018-01-22-javascript-hard-parts/fem-JavaScriptTheHardParts-ClosuresSolutions.js#L106

Hi, in the solution you are missing part of the assignment that says to pass any arguments after wait to the callback function.

### PR DESCRIPTION
**Pull request by kristakoch**

### CODE REVIEW

File: 2018-01-22-javascript-hard-parts/fem-JavaScriptTheHardParts-ClosuresSolutions.js [hunk 1]
```diff
@@ -104,8 +104,10 @@ afterCalled(); // -> 'hello' is printed
 // EXTENSION: CHALLENGE 7 //
 
 function delay(func, wait) {
-    setTimeout(func, wait);
-}
+    var args = Array.from(arguments);
+    if (args.length > 2) { args = args.slice(2, args.length) }
+    setTimeout(function() { func.apply(null,args) }, wait);
+}
```
Comment (kenearley) [inline 2018-01-22-javascript-hard-parts/fem-JavaScriptTheHardParts-ClosuresSolutions.js:110]: A possibly cleaner way to write this with destructuring would be:

```js
function delay() {
  let [func, wait, ...args] = Array.from(arguments);
  setTimeout(function() { func.apply(null, args) }, wait);
}
```

Comment (kenearley) [reply 2018-01-22-javascript-hard-parts/fem-JavaScriptTheHardParts-ClosuresSolutions.js:110]: The down side of using the destructuring is that its not as clear that `func` and `wait` are required args.

Comment (kenearley) [reply 2018-01-22-javascript-hard-parts/fem-JavaScriptTheHardParts-ClosuresSolutions.js:110]: Oh! You can do this (with the leading commas in the destructuring):

```js
function delay(func, wait) {
  let [, , ...args] = Array.from(arguments);
  setTimeout(function() { func.apply(null, args) }, wait);
}
```

**Review by kenearley — APPROVED** [review]
You solution works well. I just added some comments for possible other solutions.

**Comment by kristakoch** [comment]
Closing this because a new PR will be created to close the issue.

**Pull request closed without merging**